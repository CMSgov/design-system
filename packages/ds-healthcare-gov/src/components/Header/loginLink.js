/**
 * Returns a link pointing to the login page
 * @param {Function} translate - i18n translator
 * @param {Boolean} deConsumer - is the consumer coming from a Direct Enrollment flow?
 * @param {String} primaryDomain
 * @returns {Object}
 */
export default function (t, deConsumer, primaryDomain = '') {
  return {
    label: t('header.login'),
    href: deConsumer ? `${primaryDomain}/login?check_de=1` : `${primaryDomain}/login`,
  };
}
