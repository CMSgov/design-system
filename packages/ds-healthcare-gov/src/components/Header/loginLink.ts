import { TFunction } from 'i18next';
import { Language } from '../i18n';

/**
 * Returns a link pointing to the login page
 * @param {Function} translate - i18n translator
 * @param {Boolean} deConsumer - is the consumer coming from a Direct Enrollment flow?
 * @param {String} primaryDomain
 * @returns {Object}
 */
export default function loginLink(t: TFunction, deConsumer?: boolean, primaryDomain = '') {
  return {
    label: t('header.login'),
    href: deConsumer ? `${primaryDomain}/login?check_de=1` : `${primaryDomain}/login`,
  };
}
