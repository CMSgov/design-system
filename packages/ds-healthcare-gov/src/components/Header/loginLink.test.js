import loginLink from './loginLink';
const translate = (key) => key;

describe('loginLink', () => {
  it('returns /login link', () => {
    expect(loginLink(translate, false)).toMatchSnapshot();
  });

  it('returns link with deConsumer param', () => {
    expect(loginLink(translate, true)).toMatchSnapshot();
  });

  it('returns link with absolute URL', () => {
    expect(loginLink(translate, false, 'https://www.healthcare.gov')).toMatchSnapshot();
  });
});
