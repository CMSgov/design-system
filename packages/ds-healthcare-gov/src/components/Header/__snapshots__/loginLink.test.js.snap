// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginLink returns /login link 1`] = `
Object {
  "href": "/login",
  "label": "header.login",
}
`;

exports[`loginLink returns link with absolute URL 1`] = `
Object {
  "href": "https://www.healthcare.gov/login",
  "label": "header.login",
}
`;

exports[`loginLink returns link with deConsumer param 1`] = `
Object {
  "href": "/login?check_de=1",
  "label": "header.login",
}
`;
